main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala

```diff
@@ -249,9 +276,14 @@ object Redundancy extends Phase[TypedAst.Root, TypedAst.Root] {
       else
         Used.empty + HiddenVarSym(sym, loc)
 
-    case Expression.Def(sym, _, _, _) => Used.of(sym)
+    case Expression.Def(sym, _, _, _) =>
+      if (env0.recursionContext.exists(r => r.isRecursiveCall(sym))) {
```

Comment:
`env0.recursionContext.exists(_.isRecursiveCall(sym))`

Reply:
Oops, missed this. I'll get this before the final merge.